chore: prepare 0.46.5 release (#13816)
cosmos/cosmos-sdk: RELEASE_NOTES.md

# Cosmos SDK v0.46.5 Release Notes

This release introduces a number of bug fixes and improvements.  
Notably, an upgrade to Tendermint [v0.34.23](https://github.com/tendermint/tendermint/releases/tag/v0.34.23).

Please see the [CHANGELOG](https://github.com/cosmos/cosmos-sdk/blob/release/v0.46.x/CHANGELOG.md) for an exhaustive list of changes.

Full Commit History: https://github.com/cosmos/cosmos-sdk/compare/v0.46.4...v0.46.5

**NOTE**: The changes mentioned in `v0.46.3` are **still** required:

> ```go

Reviewer: Btw I didn't mean to block or force a suggestion, just giving my humble opinion :p

Author: Np :)

# Cosmos SDK v0.46.5 Release Notes

This release introduces a number of bug fixes and improvements.  
Notably, an upgrade to Tendermint [v0.34.23](https://github.com/tendermint/tendermint/releases/tag/v0.34.23).

> Moreover, for chains using coin metadata, every single `v0.46.x` release, prior to `v0.46.5`, **must not be used**.  
> This is due to an issue in the coin metadata migration. This is issue is fixed in `v0.46.5`.  
> When already using `<= v0.46.5` and coin metadata, a **coordinated upgrade** to `v0.46.5` is required.

Reviewer 2: Proposing another formulation:

```suggestion
If your chain's state has coin metadata, an issue has been discovered in the coin metadata migration. This issue is fixed in `v0.46.5`.  
- If you are planning to migrate to v0.46, please use `v0.46.5`. All releases prior to `v0.46.5`, **must not be used**.
- If your chain is already on v0.46 using `<= v0.46.4` and has coin metadata, a **coordinated upgrade** to `v0.46.5` is required.
```

Please see the [CHANGELOG](https://github.com/cosmos/cosmos-sdk/blob/release/v0.46.x/CHANGELOG.md) for an exhaustive list of changes.

Full Commit History: https://github.com/cosmos/cosmos-sdk/compare/v0.46.4...v0.46.5

> **NOTE**: The changes mentioned in `v0.46.3` are **still** required:

Reviewer 2: ```suggestion
**NOTE**: The changes mentioned in `v0.46.4` are **still** required:
```

Author: Dragonberry is 0.46.3, so it is v0.46.3

# Cosmos SDK v0.46.5 Release Notes

This release introduces a number of bug fixes and improvements.  
Notably, an upgrade to Tendermint [v0.34.23](https://github.com/tendermint/tendermint/releases/tag/v0.34.23).

If your chain's state has coin metadata, an issue has been discovered in the coin metadata migration. This issue is fixed in `v0.46.5`.  

> * If you are planning to migrate to v0.46, please use `v0.46.5`. All releases prior to `v0.46.5`, **must not be used**. All previous version of `v0.46` are retracted.

Reviewer: ```suggestion
* If you are planning to migrate to v0.46, please use `v0.46.5`. All releases prior to `v0.46.5`, **must NOT be used**. All previous version of `v0.46` are retracted.
```

* If your chain is already on v0.46 using `<=v0.46.4` and has coin metadata, a **coordinated upgrade** to `v0.46.5` is required.

> * If your chain is already on v0.46 using `<=v0.46.4` but has no coin metadata, this release is **not breaking**.

Author: ```suggestion
* If your chain is already on v0.46 using `<=v0.46.4` but has no coin metadata, this release is **non-breaking**.
```
